src/tasks/block/cfg.rs

```diff
@@ -1,23 +1,90 @@
-//! This file implements the [`trevm::Cfg`] and [`trevm::Block`] traits for Pecorino blocks.
+//! This file implements the [`trevm::Cfg`] and [`trevm::Block`] traits for Signet and host networks.
+use reth_chainspec::ChainSpec;
+use signet_block_processor::revm_spec;
+use signet_constants::{mainnet, pecorino};
+use signet_genesis::{
+    MAINNET_GENESIS, MAINNET_HOST_GENESIS, PECORINO_GENESIS, PECORINO_HOST_GENESIS,
+};
+use std::sync::OnceLock;
 use trevm::revm::{context::CfgEnv, primitives::hardfork::SpecId};
 
-/// PecorinoCfg holds network-level configuration values.
+/// The RU [`ChainSpec`].
+static RU_SPEC: OnceLock<SpecId> = OnceLock::new();
```

Comment:
i think you have misread an earlier comment. we don't want to memozie the `SpecId` as it may change over the course of normal operation. if we memoize the `SpecId` the builder will malfunction at hard fork blocks

we want to memoize the `ChainSpec` for the chains